### PR TITLE
PP-5520 Allow overriding of gateway account id on search

### PR DESCRIPTION
**Pull request by mrlumbu**
Use the already existing AccountIdSupplierManager to manage access
to transaction search without supplying a gateway account id. This
can be used by internal tools to get cross account data.

I have tried to minimise changes to make it clear what I have done, but
I have also started to move TransactionSearchParams towards a more
flexible structure. I have not over tested the account id override mechanism
as unit tests already exist for that.

### CODE REVIEW

File: src/main/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParams.java [hunk 1]
```diff
@@ -140,53 +139,59 @@ public void setDisplaySize(Long displaySize) {
         }
     }
 
-    public String generateQuery() {
-        StringBuilder sb = new StringBuilder();
+    public List<String> getFilterTemplates() {
+        List<String> filters = new ArrayList<>();
```
Comment (sfount) [inline src/main/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParams.java]: 👍meaningful naming

File: src/main/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParams.java [hunk 2]
```diff
@@ -254,48 +259,53 @@ public TransactionType getTransactionType() {
...
-            sb.append("&" + TRANSACTION_TYPE_FIELD + "=" + transactionType);
+            queries.add(TRANSACTION_TYPE_FIELD + "=" + transactionType);
         }
+        queries.add("page=" + forPage);
+        queries.add("display_size=" + displaySize);
 
-        sb.append(addPaginationParams(forPage));
-        return sb.toString();
+
+        return String.join("&", queries);
```
Comment (sfount) [inline src/main/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParams.java]: 👍 non-as-fiddly queries

File: src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java [hunk 3]
```diff
@@ -173,18 +174,27 @@ public TransactionDao(Jdbi jdbi) {
 
     public Long getTotalForSearch(TransactionSearchParams searchParams) {
         return jdbi.withHandle(handle -> {
-            Query query = getQuery(searchParams, handle, SEARCH_COUNT_QUERY_STRING);
+            Query query = handle.configure(SqlStatements.class, stmts -> stmts.setUnusedBindingAllowed(true))
```
Comment (sfount) [inline src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java]: Is there a reason this `setUnusedBindingAllowed` didn't need to be set before?

Comment (mrlumbu) [reply src/main/java/uk/gov/pay/ledger/transaction/dao/TransactionDao.java]: there was, but I agree it's ugly and potentially dangerous, so have found a way of not needing this anymore

**Pull request merged**